tests/test_connection.py

```diff
@@ -3522,11 +3522,19 @@ def test_upgrade_statickey(node_factory, executor):
 @pytest.mark.developer("dev-force-features required")
 def test_upgrade_statickey_onchaind(node_factory, executor, bitcoind):
     """We test penalty before/after, and unilateral before/after"""
-    l1, l2 = node_factory.line_graph(2, opts=[{'may_reconnect': True,
-                                               'dev-force-features': ["-13", "-21"],
-                                               # We try to cheat!
-                                               'allow_broken_log': True},
-                                              {'may_reconnect': True}])
+    l1, l2 = node_factory.line_graph(2, opts=[
+        {
+            'may_reconnect': True,
+            'dev-force-features': ["-13", "-21"],
+            # We try to cheat!
+            'allow_broken_log': True,
+            'allow_warning': True,
+        },
+        {
+            'may_reconnect': True,
+            'allow_warning': True,
+        }
+    ])
```

Comment:
This commit is lazy :(

You didn't specify *why* this fixes the problem, or what it is! Your should quote the watching from the logs in the commit msg. You also reformatted making it harder to tell what changes. I fixed the same flake in one of my PRs, not sure if I did it the same way though...

Reply:
Last two commits here: https://github.com/ElementsProject/lightning/pull/4804/commits

Yes I should have made that one commit, but feel free to steal it :)

Reply:
Indeed your changes are more focused. My interpretation of the issue was that we're getting a warning from the peer because the channel is being closed, and since we allow auto reconnects this can't be avoided. `allow_broken` but then not `allow_warning` sounded funny to me.